test/types/models.test.ts

```diff
@@ -193,7 +193,9 @@ projectSchema.post('save', function() {
 });
 
 projectSchema.pre('deleteOne', function() {
```

Comment:
Actually, I took a closer look and this test is correct. `pre('deleteOne')` registers query middleware by default, not document middleware. `pre('deleteOne', { document: true })` would register document and query middleware. Can you please undo this change and see if you can get your changes to work without this change?